p2p/src/simulated/network.rs

```diff
@@ -169,9 +173,24 @@ impl<E: RNetwork + Spawner + Rng + Clock + Metrics, P: PublicKey> Network<E, P>
         }
 
         match message {
```

Comment:
We should drop messages before attempting to send if the recipient is not in any peer set (even if there is an explicit link)